### ISSUE TITLE
4W Grid improvements

### ISSUE THREAD

**Issue by IanMayo** [debrief/debrief#1762]
````
4W grid won’t let me input a Xmax > “AZ”, I need one to go to at least BK.
 
I also can’t change font color separately from line color, or change line style.
 
I like the 4W functionality because it dynamically labels the X/Y axis even 
as you zoom and pan. Useful for ex planning.
 ````

**Comment by IanMayo** [comment]
Initial implementation working:
![grid4w](https://cloud.githubusercontent.com/assets/1108513/15504353/7320d158-21b6-11e6-9a01-a773feffbe3b.png)

But, can't get LineStyle to be acknowledged in the render cycle.

**Comment by IanMayo** [comment]
Re-opening, to acknowledge that line-style isn't working.

**Comment by IanMayo** [comment]
Assigning this rendering issue to you, please @sambireddyt

**Comment by sambireddyt** [comment]
@IanMayo I got latest updates from master. I see line-style works as expected.
![linestyle](https://cloud.githubusercontent.com/assets/1883173/15564319/f7f4e6d4-22de-11e6-8fd9-0f133009ae76.png)

**Comment by sambireddyt** [comment]
@IanMayo : any update? I think it is working for me locally.

**Comment by IanMayo** [comment]
@sambireddyt - are you running MS Windows?

It's not fixed on my Linux (though MS Windows is the most important target).

**Comment by sambireddyt** [comment]
@IanMayo : yes on Windows.

**Comment by IanMayo** [comment]
Ok. The line style issue isn't present on MS Windows.

It's present on Linux. We need to find out if it's present on OSX.

@theanuradha - sometime, would you mind checking if this line-style issue is present on OSX?

The steps to reproduce are at the start of this issue.

**Comment by theanuradha** [comment]
@IanMayo I get nothing on enabling 4W grid

![image](https://cloud.githubusercontent.com/assets/535713/17655928/afc013f6-62d1-11e6-9f11-12b8935aa751.png)

**Comment by theanuradha** [comment]
@IanMayo I put debug on **Grid4WPainter** and it get called I did zoom-out fully still did not get the grid. I'm i missing something ?

**Comment by IanMayo** [comment]
Ok, it's way, way off down to right.  As the instructions say, you have to zoom out, then drag it back up the working area. Here is how far you have to zoom out:
![tactical analysis - org mwc cmap combined feature-root_installs-sample_data-sample dpf - debriefng _005](https://cloud.githubusercontent.com/assets/1108513/17659995/515ab350-62ce-11e6-9d77-d8237414c68b.png)

**Comment by theanuradha** [comment]
![image](https://cloud.githubusercontent.com/assets/535713/17689277/15c92002-63a4-11e6-9af0-ef907c8520e1.png)
@IanMayo works grat on OSX as I have linux setup I can debug there. What do you think ?

**Comment by IanMayo** [comment]
Aah yes, I think we're getting a couple of problems associated with line styles not working on all O/Ss.

I can see it's not showing as dotted in Ubuntu:
![linestyle](https://cloud.githubusercontent.com/assets/1108513/17691430/5361e11e-638c-11e6-8d46-1e17a0f12719.png)

**Comment by theanuradha** [comment]
@IanMayo I can try to debug in linux as I have it setup

**Comment by IanMayo** [comment]
Great - it's all yours.

**Comment by theanuradha** [comment]
@IanMayo as of current investigation/debug line style not at all working **SWTCanvasAdapter**
Looking in to this 

![image](https://cloud.githubusercontent.com/assets/535713/17748274/365ad026-64d6-11e6-8478-c8a80b68a1f7.png)

**Comment by IanMayo** [comment]
Line Style for a track may have broken logic.    Performance optimisations in the TrackWrapper paint algorithms have left very complex code.

Let's concentrate on the 4W Grid line styles for now.

**Comment by theanuradha** [comment]
@IanMayo yes Im focus on 4W Grid , It correctly pass style to **SWTCanvasAdapter** it is just going wrong after that.

**Comment by IanMayo** [comment]
Just in Linux?  Can you carry on drilling down? Are there any related Eclipse bugs?

**Comment by theanuradha** [comment]
It is bit strange as I have demo SWT app with Direct Drawing on Canvas using PaintEvent.gc and works fine 


![image](https://cloud.githubusercontent.com/assets/535713/17748483/3efa77f8-64d7-11e6-8109-c5c3f57d5913.png)

**Comment by theanuradha** [comment]
@IanMayo it is just in linux I did search on Linux with eclipse bug repo but no results on subject.

Only issue/difference we have with **PaintEvent.gc** vs PlotEditor create GC object with Image 
   ```
_dblBuff = new Image(Display.getCurrent(), theSize.x, theSize.y);
      final GC theDest = new GC(_dblBuff);
```

**Comment by theanuradha** [comment]
@IanMayo I have try many workaround but none works :( . do we need to invest more time ? or we can leave it for now

**Comment by IanMayo** [comment]
Ok. Please produce a clear statement of just what we're experiencing, across the different O/S, together with the suspected cause, and a general description of the fixes you've tried?

I can then "park" the issue, or try to invite a Fixed Price fix from a Java/Linux UI expert.
